### PR TITLE
PR for #29

### PR DESCRIPTION
**Pull request by andrestaht**
Didn't bother rewriting the tests yet because this idea might get scrapped.
Probably should make `createState` and `removeUncheckedKeys` accessible like other functions are defined in `index.js`.
With current state creating ci and update could be removed form `determineConfig.js`.

My `mochaSetup.js`:
```
chai.use(chaiJestSnapshot);

before(function() { // eslint-disable-line no-undef
    chaiJestSnapshot.resetSnapshotRegistry();
});

beforeEach(function() {
    // Save snapshots to __snapshots__ dir
    const filename = path.resolve(path.dirname(this.currentTest.file), "__snapshots__", path.basename(this.currentTest.file) + ".snap");

    chaiJestSnapshot.setFilename(filename);
    chaiJestSnapshot.setTestName(this.currentTest.fullTitle());
    chaiJestSnapshot.createState({
        snapshotPath: filename,
        ci: process.env.CI,
        update: process.env.CHAI_JEST_SNAPSHOT_UPDATE_ALL,
    });
});
after(function() { // eslint-disable-line no-undef
    chaiJestSnapshot.removeUncheckedKeys();
});
```

### CODE REVIEW

**Comment by suchipi** [comment]
Thanks, I'll take a look soon

**Comment by suchipi** [comment]
Sorry, I looked at it briefly but have not had a chance to follow back up, I apologize for the long delay.

I don't think there's a straightforward way to support both `matchSnapshot(true)` and the snapshot state from jest, so I think we should move to the latter and make this a breaking change.

File: src/index.js [hunk 1]
```diff
@@ -65,6 +66,9 @@ chaiJestSnapshot.resetSnapshotRegistry = function resetSnapshotRegistry() {
   }
```
Comment (suchipi) [inline src/index.js]: Could you add the stuff you had in your beforeEach to `configureUsingMochaContext`?

File: src/index.js [hunk 2]
```diff
@@ -65,6 +66,9 @@ chaiJestSnapshot.resetSnapshotRegistry = function resetSnapshotRegistry() {
...
 }
 
+chaiJestSnapshot.createState = snapshotStateHandler.create;
+chaiJestSnapshot.removeUncheckedKeys = snapshotStateHandler.removeUncheckedKeys;
```
Comment (suchipi) [inline src/index.js]: We will have to refactor the documentation to note that you need to call these in your `beforeAll`/`afterAll`.

File: src/snapshotStateHandler.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { SnapshotState } from "jest-snapshot";
+
+let states = {};
+
+module.exports = {
+    create: ({snapshotPath, ci, update}) => {
```
Comment (suchipi) [inline src/snapshotStateHandler.js]: The `ci` and `update` arguments can stay internal here (hardcoded to the process environment variables); the public interface should only take the path

Comment (andrestaht) [reply src/snapshotStateHandler.js]: Good idea! :)

**Comment by andrestaht** [comment]
Thanks for the review!
Pushed some changes.

There are still couple of things missing:
1. `snapshotStateHandler` isn't covered with tests
2. ci and update aren't tested in `matchSnapshot`

**Comment by andrestaht** [comment]
@suchipi could you take another look at this?
Didn't find a good way to test ci and update flags within matchSnapshot tests but added some unit tests for snapshotStateHandler.

**Comment by suchipi** [comment]
Hey, sorry for the radio silence on this. I will try to take a look soon.

**Comment by suchipi** [comment]
Hey, I'm sorry this is still open. I took a look 22 days ago when you commented and started working through behavior and thinking about how it should work, but I didn't have time to finish. I'll let you know when I look at it again- sorry again and thank you for being so patient 🙏

**Comment by suchipi** [comment]
@andrestaht are you interested in me giving you write access to this repo and you being a co-maintainer of the package?

**Comment by andrestaht** [comment]
@suchipi why not :)

**Comment by mAAdhaTTah** [comment]
I'm interested in this feature as well. Anything I can do to move this along?

**Comment by fa93hws** [comment]
Any update?

**Comment by andrestaht** [comment]
Sorry for the silence. I switched companies and this dependency is no longer part of the stack.

If someone still sees value in this then feel free to take it over.

I'll help with the review and getting the PR released.

Sorry again for this long unactivity.

**Pull request closed without merging**